src/renderer/components/ft-playlist-add-video-prompt/ft-playlist-add-video-prompt.js

```diff
@@ -238,10 +269,16 @@ export default defineComponent({
         const playlist = this.allPlaylists.find((list) => list._id === selectedPlaylistId)
         if (playlist == null) { return }
 
+        // Use [].concat to avoid `do not mutate vuex store state outside mutation handlers`
+        let videosToBeAdded = [].concat(this.toBeAddedToPlaylistVideoList)
+        if (!this.addingDuplicateVideosEnabled) {
+          const playlistVideoIds = playlist.videos.map((v) => v.videoId)
+          videosToBeAdded = videosToBeAdded.filter((v) => !playlistVideoIds.includes(v.videoId))
```

Comment:
**suggestion (non-blocking):** Would using a `Set` here be more performant?

Reply:
I wish there is a linter to provide warnings...

Reply:
What would that linting rule be? Put a warning on all types of array searches? Data structure choice is always going to be up to developer discernment, I'd imagine

Reply:
True, Set can't have duplicate values, ignore~